Lib/test/test_pty.py

```diff
@@ -81,28 +101,74 @@ def setUp(self):
     def handle_sig(self, sig, frame):
         self.fail("isatty hung")
 
+    # RELEVANT ANYMORE?
     @staticmethod
     def handle_sighup(signum, frame):
         # bpo-38547: if the process is the session leader, os.close(master_fd)
         # of "master_fd, slave_name = pty.master_open()" raises SIGHUP
         # signal: just ignore the signal.
         pass
 
-    def test_basic(self):
+    @expectedFailureIfStdinIsTTY
+    def test_openpty(self):
+        try:
+            mode = tty.tcgetattr(pty.STDIN_FILENO)
+        except tty.error:
+            # pty.STDIN_FILENO not a tty?
+            debug("tty.tcgetattr(pty.STDIN_FILENO) failed")
+            mode = None
+
+        try:
+            TIOCGWINSZ = tty.TIOCGWINSZ
+            TIOCSWINSZ = tty.TIOCSWINSZ
+        except AttributeError:
+            debug("TIOCSWINSZ/TIOCGWINSZ not available")
+            winsz = None
+        else:
+            try:
+                debug("Setting pty.STDIN_FILENO window size")
+                current_stdin_winsz = os.get_terminal_size(pty.STDIN_FILENO)
+
+                # Set number of columns and rows to be the
+                # floors of 1/5 of respective original values
+                winsz = struct.pack("HHHH", current_stdin_winsz.lines//5,
+                                    current_stdin_winsz.columns//5, 0, 0)
+                fcntl.ioctl(pty.STDIN_FILENO, TIOCSWINSZ, winsz)
+
+                # Were we able to set the window size
+                # of pty.STDIN_FILENO successfully?
+                s = struct.pack("HHHH", 0, 0, 0, 0)
+                new_stdin_winsz = fcntl.ioctl(pty.STDIN_FILENO, TIOCGWINSZ, s)
+                self.assertEqual(new_stdin_winsz, winsz,
+                                 "pty.STDIN_FILENO window size unchanged")
+            except OSError:
+                # pty.STDIN_FILENO not a tty?
+                debug("Failed to set pty.STDIN_FILENO window size")
```

Comment:
You are absolutely correct. I will change this. I have one question before I change this. If I use warnings.warn() here, then it will always generate warnings when the test is not run with stdin being a tty; for example the automated tests done by GitHub. Is that going to be an issue?

Reply:
It may cause a slight bit of extra noise on the tests, but it shouldn't cause the GitHub CI to fail when the warning is emitted (I don't believe we have any of our CI set to fail on warning, but it can be ran with that setting locally).